lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetWriters.java

```diff
@@ -202,6 +212,9 @@ private static PrimitiveValueWriter getValueWriter(ValuesWriter valuesWriter, Ty
             verifyParquetType(type, parquetType, OriginalType.TIME_MICROS);
             return new TimeMicrosValueWriter(valuesWriter, parquetType);
         }
+        if (type instanceof TimestampType && parquetType.getPrimitiveTypeName().equals(PrimitiveType.PrimitiveTypeName.INT96)) {
```

Comment:
the following 3 checks within this class are also related to checking whether we're dealing with a `Timestamp` type:
https://github.com/trinodb/trino/pull/12857/files#diff-3b64954c48d2c0059a9156e9ef84a6b66d58c436ea593753a78cd27a604c92feR218-R233



WDYT about structuring the code like this:

```
if (type instanceof TimestampType) {
    if (parquetType.getPrimitiveTypeName().equals(PrimitiveType.PrimitiveTypeName.INT96)) {
      ....
    }
    if (TIMESTAMP_MILLIS.equals(type)) {
       ....
    }
    if (TIMESTAMP_MICROS.equals(type)) {
        ....
    }
    if (TIMESTAMP_NANOS.equals(type)) {
         ....
     }
}

```

Reply:
sounds good, made the change